### ISSUE TITLE
Batch size mismatch in node-classification with multiple MP layers

### ISSUE THREAD

**Issue by xiaohan2012** [xiaohan2012/stgym#68]
## Parent Issue
Follow-up from #48 (experiment `960518417917991538` analysis)

## Problem
16 runs fail with a batch size mismatch error in node-classification tasks when using multiple message-passing layers. The model output has the pooled node count as its batch dimension, while the target retains the original node count.

## Error Message
```
Expected input batch_size (X) to match target batch_size (Y)
```
Where X is the pooled size (20, 40, 80) and Y is the original node count (1475, 3583, 7882, 8025, 12132, 42246, 45465, 52118).

## Statistics

**16 failures** (31.4% of all 51 failures)

### By Dataset
| Dataset | Count |
|---------|-------|
| colorectal-cancer | 10 |
| charville | 2 |
| human-crc | 2 |

### By Design Dimension
All 16 are from `model.num_mp_layers`.

### By Task Type
All 16 are **node-classification** — this error does not occur in graph-classification.

### Run `23cd0b8d` (human-crc, 2 MP layers)
```
dataset: human-crc
task: node-classification
model/mp_layers/0/layer_type: gcnconv
model/mp_layers/0/dim_inner: 32
model/mp_layers/0/pooling/type: mincut
model/mp_layers/0/pooling/n_clusters: 10
model/mp_layers/0/normalize_adj: True
model/mp_layers/0/use_batchnorm: False
model/mp_layers/1/layer_type: gcnconv
model/mp_layers/1/dim_inner: 32
model/mp_layers/1/pooling/type: mincut
model/mp_layers/1/pooling/n_clusters: 10
model/post_mp_layer/dims: [32, 16]
train/optim/base_lr: 0.1
data_loader/graph_const: knn
data_loader/knn_k: 20
```
Error: `Expected input batch_size (80) to match target batch_size (3583)`

## Likely Root Cause
In node-classification, the loss function expects per-node predictions matching the original graph. However, hierarchical pooling (DMoN/MinCut) coarsens the graph, reducing the node count. With multiple MP+pooling layers, the output node count shrinks further. The model produces predictions for `n_clusters` pooled nodes but the target has the original `N` nodes.

This is architecturally incompatible: hierarchical pooling is designed for graph-level tasks, not node-level tasks where per-node outputs are needed.

## MLflow Run IDs
<details>
<summary>All 16 run IDs</summary>

- `23cd0b8d82ec44b4be2363b320607366` — human-crc (batch 80 vs 3583)
- `591a5aa2a08347b785e10545028313c8` — human-crc (batch 80 vs 12132)
- `39183654e9494ec391fa02d728d1284a` — charville (batch 40 vs 8025)
- `86f109fca009490dbf4f1ae6aefceddd` — charville (batch 40 vs 7882)
- `25623543a75a48a08da0c775b331d462` — colorectal-cancer (batch 20 vs 42246)
- `69f32972c0904252b4eb8ce1023b98e6` — colorectal-cancer (batch 20 vs 45465)
- `82aed3395bb1492da75107c930847b6f` — colorectal-cancer (batch 20 vs 45465)
- `896b6b98e10d4d20988e7ea0673d5777` — colorectal-cancer (batch 20 vs 52118)
- `8f5660b828fd4fa7bd00af3fdd13dfe4` — colorectal-cancer (batch 20 vs 42246)
- `9dd9eb16897240b2af5239206151a7bd` — colorectal-cancer (batch 20 vs 52118)
- `f4e15537bd724d68b2575d8ddde77b09` — colorectal-cancer (batch 20 vs 1475)
- `fea994b521f844cdabf78c5896527cdc` — colorectal-cancer (batch 20 vs 1475)
</details>

## MLflow Experiment
`960518417917991538` on `http://127.0.0.1:5001`
